### PR TITLE
expand README files for examples

### PR DESCRIPTION
**Pull request by khorne3**
Currently features the following so that no one sees empty files:

- How to use it
- Any gotchas

I'm open to any and all changes suggested. I'm tracking https://github.com/coder/coder/issues/1321 for future additions regarding troubleshooting

### CODE REVIEW

File: examples/templates/docker/README.md [hunk 1]
```diff
@@ -6,19 +6,22 @@ tags: [local, docker]
...
+the template to make the image available to users:
 
 ```sh
+# Open the template
 vim main.tf
 ```
+
+Modify your file to match the following:
+
+```dockerfile
```
Comment (bpmct) [inline examples/templates/docker/README.md:24]: I'm using GitHub's "diff" view to indicate the changes made. If you'd prefer not to use it, this is HCL

```suggestion
```diff
```

or 

```suggestion
```hcl
```

File: examples/templates/gcp-linux/README.md [hunk 2]
```diff
@@ -3,3 +3,35 @@ name: Develop in Linux on Google Cloud
 description: Get started with Linux development on Google Cloud.
 tags: [cloud, google]
 ---
+
+# gcp-linux
```
Comment (bpmct) [inline examples/templates/gcp-linux/README.md:7]: Nice!!!

**Review by bpmct — APPROVED** [review]
Nice, thanks for doing this! This is a larger question beyond this PR, but:

1) do you have thoughts on how the product can refer to these docs during template creation? We have found that in many cases, we want to recommend users make edits to the TF instead of using the built-in example to manage access/secrets in their preferred way.
2) I noticed some of this info is repeated based on the cloud configuration, not the template itself. Perhaps not a big deal yet, but we may be able to find another solution once we build a static docs site

**Comment by khorne3** [comment]
@bpmct let me think a bit re: your questions and I'll get back to you!

File: examples/templates/do-linux/README.md [hunk 3]
```diff
@@ -6,12 +6,18 @@ tags: [cloud, digitalocean]
 
 # do-linux
```
[the remark this answers is not in the document]
Comment (khorne3) [reply examples/templates/do-linux/README.md:8]: I definitely agree that consistency is key -- after thinking about it, what do you think about removing 'em or just saying, "This template is for Ubuntu users on AWS" or similar as a brief intro so you know what you're getting into?

**Comment by khorne3** [comment]
Re: #1: I think some of the usage instructions are too long for us to consistently populate to the CLI, but maybe a link to instructions. I don't know if this is a bit too janky

Re: #2: yeah, if we could find a way to use templating to populate in repeated info, that would be really nice

**Review by bpmct — APPROVED** [review]

**Pull request merged**